arbnode/seq_coordinator.go

```diff
@@ -663,8 +664,9 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
 	}
 
 	// update wanting the lockout
+	// Sequencer should want lockout if and only if- its synced, not avoiding lockout and processedMessages is not lagging too much behind localMsgCount
 	var wantsLockoutErr error
-	if synced && !c.AvoidingLockout() {
+	if synced && !c.AvoidingLockout() && processedMessages+1 >= c.sync.SyncTargetMessageCount() {
```

Comment:
If the processedMessages is behind the SyncTargetMessageCount, why is it still marked as synced?

Reply:
Doesn't it check the execution side's sync status, or is it the other way around where the execution side checks the consensus side's sync status? Either way, I'd prefer to be checking this via the sync monitors instead of having a different check here.

Reply:
yep, its the other way around. Do you recommend we have this check as part of current `Synced()` or have a new function like `IsExecutionSynced()`?

Reply:
I think we should just use the execution's SyncMonitor instead of the consensus side's Synced() method above when determining the `synced` variable, and that way it'll check both, but I'd make sure that makes sense to @tsahee first

Reply:
Yeah, that might make more sense.

The way I'd do it is to dd Synced() and FullSyncProgressMap() functions .. I think to ExecutionSequencer(?)
Give SeqCoordinator a pointer to FullExecutionClient instead of to SyncMonitor

Reply:
Yep, ExecutionSequencer sounds good to me and also we don't need to add `Synced` and `FullSyncProgressMap` to the interface, instead we'll just add `SyncProgressMap` since `Synced` of execution syncmonitor uses `SyncProgressMap` anyway

Reply:
Why not just Synced? Isn't Synced == len(SyncProgressMap) == 0?

Reply:
in case its not synced the seq-coordinator logs the SyncProgressMap()